custom/icds_reports/migrations/sql_templates/database_functions/aggregate_awc_data.sql

```diff
@@ -496,10 +498,12 @@ BEGIN
     -- could possibly add multicol indexes to make order by faster?
 
   EXECUTE 'UPDATE ' || quote_ident(_tablename5) || ' agg_awc SET ' ||
-    'num_awc_infra_last_update = 1 WHERE infra_last_update_date IS NOT NULL';
+    'num_awc_infra_last_update = 1 WHERE infra_last_update_date IS NOT NULL AND ' ||
+     quote_literal(_month_start_6m) || ' < infra_last_update_date';
 
   EXECUTE 'UPDATE ' || quote_ident(_tablename5) || ' agg_awc SET ' ||
-    'num_awc_infra_last_update = 0 WHERE infra_last_update_date IS NULL';
+    'num_awc_infra_last_update = 0 WHERE infra_last_update_date IS NULL OR ' ||
+     quote_literal(_month_start_6m) || ' >= infra_last_update_date';
```

Comment:
Would it work to change the two separate UPDATES  to something like:

```
UPDATE table SET
  num_awc_infra_last_update = CASE WHEN infra_last_update_date IS NOT NULL AND 6 months ago < infra_last_update_date THEN 1 ELSE 0 END
```

Reply:
I have updated that. Thanks, You are right as this shows that indeed it is updating one column.